src/Core/ScreenshotCache.vala

```diff
@@ -0,0 +1,273 @@
+public class AppCenterCore.ScreenshotCache {
+    private const int MAX_CACHE_SIZE = 100000000;
+
+    public string screenshot_path;
+    private Soup.Session session;
+
+    public static ScreenshotCache? new_cache () {
```

Comment:
Why static instead of just making a new class? It's not really a singleton anyways :)

Reply:
edit: Instead of just using `new` to make an instance of the cache? :)

Reply:
The `static` keyword allows the method to be called without requiring an instance to be constructed first. After all, this is seeking to create a new instance, not use an existing one. There's a chance that the cache can't be created, so this will avoid allocating the object if the object is going to be invalid anyway.

As for singletons in general, I'm strongly opposed to their use in software. It's just bad practice.